examples/QIR/Simulation/StateSimulator/README.md

```diff
@@ -0,0 +1,375 @@
+There are only two possible results for such a measurement, given by a positive (+) and negative (-) parity, since each individual Pauli measurement returns either +1 or -1.
+Thus, the two projective measurement operators are given by `P_+- = (1 +- P_1⊗P_2⊗..⊗P_n)/2`:
+
+```cpp
+Result StateSimulator::Measure(long numBases, PauliId bases[], long numTargets, Qubit targets[])
+{
+    assert(numBases == numTargets);
+    short dim = this->numActiveQubits;
+
+    // Projection operators P_+- for Pauli measurements {P_i}:
+    //     P_+- = (1 +- P_1⊗P_2⊗..⊗P_n)/2
+    Operator paulis = BuildPauliUnitary(numTargets, bases, targets);
+    Operator p_projector = (Operator::Identity(dim, dim) + paulis)/2;
+    Operator m_projector = (Operator::Identity(dim, dim) - paulis)/2;
+
+    // Probability of getting outcome Zero is p(+) = 〈Ψ|P_+|Ψ⟩.
+    double probZero = real(this->stateVec.conjugate().dot(p_projector*this->stateVec));
+
+    // Select measurement outcome via PRNG.
+    double random0to1 = (double) rand() / (RAND_MAX);
+    Result outcome = random0to1 < probZero ? UseZero() : UseOne();
+
+    // Update state vector with |Ψ'⟩ = 1/√p(m) P_m|Ψ⟩.
+    if (outcome == UseZero())
+        this->stateVec = (p_projector * this->stateVec)/sqrt(probZero);
+    else
+        this->stateVec = (m_projector * this->stateVec)/sqrt(1-probZero);
+
+    return outcome;
+}
+```
+
+The function `BuildPauliUnitary` simply generates the `Operator` "`P_1⊗P_2⊗..⊗P_n`" over the active qubit space.
+
+
+## Compiling the simulator
+
+Download the latest stable release of the [Eigen library](http://eigen.tuxfamily.org/) and copy over the relevant headers as described in [include/Eigen](include/Eigen/README.md).
+
+Although the [QIR Runtime header files](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/public) are sufficient to compile the simulator, actually running it will require the Runtime binaries.
+Use the NuGet CLI with the commands below to download the [QIR Runtime package](https://www.nuget.org/packages/Microsoft.Quantum.Qir.Runtime) and extract the appropriate headers and libraries (adjusting the package version as required):
+
+- **Windows**:
+
+    ```shell
+    mkdir build
+    curl https://dist.nuget.org/win-x86-commandline/latest/nuget.exe --output build/nuget.exe
+    build/nuget install Microsoft.Quantum.Qir.Runtime -Version 0.18.2106148911-alpha -DirectDownload -DependencyVersion Ignore -OutputDirectory tmp
+    cp tmp/Microsoft.Quantum.Qir.Runtime.0.18.2106148911-alpha/runtimes/any/native/include/* build
+    cp tmp/Microsoft.Quantum.Qir.Runtime.0.18.2106148911-alpha/runtimes/win-x64/native/* build
+    rm -r tmp
+    ```
+
+- **Linux** (installs mono for the NuGet CLI):
+
+    ```shell
+    mkdir build
+    sudo apt update && sudo apt install -y mono-complete
+    curl https://dist.nuget.org/win-x86-commandline/latest/nuget.exe --output build/nuget
```

Comment:
The steps to download nuget and setup via mono can actually be simplified since mono-based nuget is it's own package. See for example:
https://github.com/microsoft/qsharp-compiler/blob/edd7caadca25f8deb556f33c80a6755c6f2a5808/.devcontainer/Dockerfile#L9